scio/src/main/scala/org/apache/zeppelin/scio/DisplaySCollectionImplicits.scala

```diff
@@ -0,0 +1,105 @@
+
+import scala.reflect.ClassTag
+
+/**
+ * Implicit Zeppelin display helpers for SCollection.
+ */
+object DisplaySCollectionImplicits {
+
+  private def materialize[T: ClassTag](self: SCollection[T]) = {
+    val f = self.materialize
+    self.context.close()
+    f
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinSCollection[T: ClassTag](private val self: SCollection[T])(implicit ev: T <:< AnyVal) extends AnyVal {
+  implicit class ZeppelinSCollection[T: ClassTag](val self: SCollection[T])
+                                                 (implicit ev: T <:< AnyVal) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from SCollection. */
+    def closeAndDisplay(printer: (T) => String = (e: T) => e.toString): Unit = {
+      DisplayTapImplicits.ZeppelinTap(materialize(self).waitForResult()).display(printer)
+    }
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinSCollection[T: ClassTag](private val self: SCollection[T]) extends AnyVal {
+  implicit class ZeppelinStringSCollection[T: ClassTag](val self: SCollection[T])
+                                                       (implicit ev: T <:< String) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from SCollection. */
+    def closeAndDisplay(printer: (T) => String = (e: T) => e.toString): Unit = {
+      DisplayTapImplicits.ZeppelinStringTap(materialize(self).waitForResult()).display(printer)
+    }
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinKVSCollection[K: ClassTag, V: ClassTag](val self: SCollection[(K, V)]) extends AnyVal {
+  implicit class ZeppelinKVSCollection[K: ClassTag, V: ClassTag](val self: SCollection[(K, V)]) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from KV SCollection. */
+    def closeAndDisplay(): Unit = {
+      DisplayTapImplicits.ZeppelinKVTap(materialize(self).waitForResult()).display()
+    }
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinProductSCollection[T: ClassTag](val self: SCollection[T])(implicit ev: T <:< Product) extends AnyVal {
+  implicit class ZeppelinProductSCollection[T: ClassTag](val self: SCollection[T])
+                                                        (implicit ev: T <:< Product) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from Product like SCollection */
+    def closeAndDisplay(): Unit = {
+      DisplayTapImplicits.ZeppelinProductTap(materialize(self).waitForResult()).display()
+    }
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinAvroSCollection[T: ClassTag](val self: SCollection[T])(implicit ev: T <:< GenericRecord) extends AnyVal {
+  implicit class ZeppelinAvroSCollection[T: ClassTag](val self: SCollection[T])
+                                                     (implicit ev: T <:< GenericRecord) {
```

Comment:
Starting from `1.7.5` `SpecificRecordBase` implements `GenericRecord` (~2013). In here we use 1.7.7. That said, it might worth supporting old versions and use `IndexedRecord`.